backend/src/services/auto_trader.py

```
@@ -120,151 +109,123 @@ async def _scan_and_buy(self):
+                current_balance = await self.wallet_service.get_token_balance(token_address)
+                if current_balance <= 0:
+                    tokens_to_remove.append(token_address)
                     continue
 
+                # Sync balance if it changed
+                if current_balance != details.get('amount_tokens'):
+                    details['amount_tokens'] = current_balance
+                    save_position(details)
+
+                current_token_data = await self.data_fetcher_service.get_token_by_address(token_address)
+                if not current_token_data: continue
```

Ruff (0.15.6)

[error] Multiple statements on one line (colon)

(E701)